node/test-helpers/subsystem/src/lib.rs

```diff
@@ -15,215 +15,7 @@
-				}
-
-				if let Some(waker) = flush_waker {
-					waker.wake();
-				}
-
-				Poll::Ready(Some(item))
-			}
-		}
-	}
-}
-
-/// Create a single-item Sink/Stream pair.
-///
-/// The sink's send methods resolve at the point which the stream reads the item,
-/// not when the item is buffered.
-pub fn single_item_sink<T>() -> (SingleItemSink<T>, SingleItemStream<T>) {
-	let inner = Arc::new(Mutex::new(SinkState::Empty { read_waker: None }));
-	(
-		SingleItemSink(inner.clone()),
-		SingleItemStream(inner),
-	)
-}
-
-/// A test subsystem context.
-pub struct TestSubsystemContext<M, S> {
-	tx: mpsc::UnboundedSender<AllMessages>,
-	rx: SingleItemStream<FromOverseer<M>>,
-	spawn: S,
-}
-
-#[async_trait::async_trait]
-impl<M: Send + 'static, S: Spawn + Send + 'static> SubsystemContext for TestSubsystemContext<M, S> {
-	type Message = M;
-
-	async fn try_recv(&mut self) -> Result<Option<FromOverseer<M>>, ()> {
-		match poll!(self.rx.next()) {
-			Poll::Ready(Some(msg)) => Ok(Some(msg)),
-			Poll::Ready(None) => Err(()),
-			Poll::Pending => Ok(None),
-		}
-	}
-
-	async fn recv(&mut self) -> SubsystemResult<FromOverseer<M>> {
-		self.rx.next().await.ok_or(SubsystemError)
-	}
-
-	async fn spawn(&mut self, s: Pin<Box<dyn Future<Output = ()> + Send>>) -> SubsystemResult<()> {
-		self.spawn.spawn(s).map_err(Into::into)
-	}
-
-	async fn send_message(&mut self, msg: AllMessages) -> SubsystemResult<()> {
-		self.tx.send(msg).await.expect("test overseer no longer live");
-		Ok(())
-	}
-
-	async fn send_messages<T>(&mut self, msgs: T) -> SubsystemResult<()>
-		where T: IntoIterator<Item = AllMessages> + Send, T::IntoIter: Send
-	{
-		let mut iter = stream::iter(msgs.into_iter().map(Ok));
-		self.tx.send_all(&mut iter).await.expect("test overseer no longer live");
-
-		Ok(())
-	}
-}
-
-/// A handle for interacting with the subsystem context.
-pub struct TestSubsystemContextHandle<M> {
-	tx: SingleItemSink<FromOverseer<M>>,
-	rx: mpsc::UnboundedReceiver<AllMessages>,
-}
-
-impl<M> TestSubsystemContextHandle<M> {
-	/// Send a message or signal to the subsystem. This resolves at the point in time where the
-	/// subsystem has _read_ the message.
-	pub async fn send(&mut self, from_overseer: FromOverseer<M>) {
-		self.tx.send(from_overseer).await.expect("Test subsystem no longer live");
-	}
-
-	/// Receive the next message from the subsystem.
-	pub async fn recv(&mut self) -> AllMessages {
-		self.rx.next().await.expect("Test subsystem no longer live")
-	}
-}
-
-/// Make a test subsystem context.
-pub fn make_subsystem_context<M, S>(spawn: S)
-	-> (TestSubsystemContext<M, S>, TestSubsystemContextHandle<M>)
-{
-	let (overseer_tx, overseer_rx) = single_item_sink();
-	let (all_messages_tx, all_messages_rx) = mpsc::unbounded();
-
-	(
-		TestSubsystemContext {
-			tx: all_messages_tx,
-			rx: overseer_rx,
-			spawn,
-		},
-		TestSubsystemContextHandle {
-			tx: overseer_tx,
-			rx: all_messages_rx
-		},
-	)
-}
+pub use polkadot_subsystem::test_helpers::*;
```

Comment:
don't think this is going to work given test_helpers is `cfg(test)` gated
substrate uses features approach: https://github.com/paritytech/substrate/search?l=TOML&q=test-helpers
this is prone to accidental feature leaking ofc and needs to be used with care

Reply:
To my mind that's a feature, not a bug: it reduces the chance that anybody accidentally uses a test helper in production code.

Reply:
I meant it doesn't compile :)

Reply:
Oh, I see. Right, `cfg(test)` doesn't propagate. I'll fix that.